docs/design/kubefile_alpha.md

```diff
@@ -96,20 +96,49 @@ contains multiple yaml or a single yaml for deploying CSI. `https://...; oss://`
 
 `APP` allows user to specify which applications to be installed over Kubernetes.
 
-Some behaviors are available:
+The `APP` instruction has one form:
 
-`APP [UNIQUE-NAME] [path, https://, oss://]`. `[UNIQUE-NAME]` is a declaration for an application, which is unique for sealer image. `path` is a relative path refers to local build context, this can be a directory
-contains multiple yaml or a single file for deploying app. `https://...; oss://` are remote addresses for downloading related files to deploy applications.
+`APP APP_NAME scheme:path1 scheme:path2`.
 
-`[]` means user could declare several sources
+The `APP_NAME` is a unique name to kube image.
+
+The `scheme` has three forms:
+
+* `local://path_rel_2_build_context` (files are from build context, path is relative to build context)
+* `http://example.yaml`
+* `https://example.yaml`
+* `helm://` (unrealized)
+
+There can be many `APP` in a `Kubefile`.
 
 ### LAUNCH
 
 `LAUNCH` allows user to specify which apps(specified by instruction `APP`) to start right after the completion of cluster initiation.
 Users are able to declare which applications to launch within the sealer image.
 
-#### Example
+The `LAUNCH` instruction has one form:
 
-```
-LAUNCH --app [mysql, es, ...] --rawcmd [kubectl apply -f *.yaml, ...] --helm [...]
-```
\ No newline at end of file
+`LAUNCH APP_1 APP_2`
+
+`LAUNCH ["APP_1", "APP_2"]`
+
+There are two behaviors for `LAUNCH`:
+
+* `helm install APP_1 APP_1_PATH_REL_2_Rootfs` (APP_1 is detected as helm app)
+* `kubectl apply -f APP_2_PATH_REL_2_Rootfs` (APP_2 is detected as raw yaml set app)
+
+The behaviors will be generated automatically, users don't have to care about that.
+
+There can be only one `LAUNCH` or `CMDS` instruction in a `Kubefile`.
+
+### CMDS
```

Comment:
How about just naming as `CMD` instead of  `CMDS`? Here is my reasons:
* Telling users that Kubefile only allows users to input one command. One command could be quite strict for users to understand;
* Command format seems to be shell scripts. If end users wish to let command play several roles, users could use `&&` to combine commands;
* It is explicit that what is the running/exit status of command. If there are serveral commands, I think it becomes harder to explain what is the status of commands running.

In addition, is the CMD only allowed to execute towards k8s/k3s/k0s APIServer? Or CMD is allowed to run towards Linux Kernel, which seems weird? After all we are cluster application oriented.

Reply:
I'm ok with `CMD` or `CMDS`.
I think it's fine to define several cmds in `CMD` like `CMD ["kubectl ...", "kubectl ..."]`. And of course user can do like `CMD ["kubectl xxx && kubectl xxxxx"]`.

`CMD` will exec over master0 machine, so it can be executed towards apiserver and Linux kernel both. IMO, `CMD` is just an reserved instruction, which is not recommended.